inginious/frontend/templates/custom_values.html

```diff
@@ -0,0 +1,90 @@
+                {% else %}
+                    {{ _("Group management") }}
+                {% endif %}
+            </a>
+            {% if course.allow_unregister() %}
+                <a href="#" data-toggle="modal" data-target="#unregister_modal" class="list-group-item list-group-item-action list-group-item-info">
+                    <i class="fa fa-user-times fa-fw"></i>&nbsp; {{ _("Unregister from this course") }}
+                </a>
+            {% endif %}
+            <a class="list-group-item list-group-item-action list-group-item-info" href="{{ get_homepath() }}/custom/{{ course.get_id() }}">
+                <i class="fa fa-cogs fa-fw"></i>&nbsp; {{ _("Custom fields") }}
+            </a>
+        </div>
+    {% elif not user_manager.session_logged_in() %}
+        {% include "signin_button.html" %}
+    {% elif course.is_registration_possible(user_info) %}
+        <h3>{{ _("Enroll in the course") }}</h3>
+
+        <div class="alert alert-warning" role="alert">
+            {{ _("Please enroll in the course to be able to submit answers to problems.") }}
+        </div>
+        <div class="list-group">
+            <a href="{{ get_homepath() }}/register/{{ course.get_id() }}" class="list-group-item list-group-item-action list-group-item-info">
+                <i class="fa fa-sign-in fa-fw"></i>&nbsp; {{ _("Enroll in the course") }}
+            </a>
+        </div>
+    {%endif %}
+{% endblock %}
+{% block navbar %}
+    <nav aria-label="breadcrumb">
+        <ol class="breadcrumb">
+            {% if registered %}
+                <li class="breadcrumb-item"><a href="{{ get_homepath() }}/mycourses" title="{{ _('My courses') }}" data-toggle="tooltip" data-placement="bottom"><i class="fa fa-th-list"></i></a></li>
+            {% else %}
+                <li class="breadcrumb-item"><a href="{{get_homepath() }}/courselist" title="{{ _('Course list') }}" data-toggle="tooltip" data-placement="bottom"><i class="fa fa-th-list"></i></a></li>
+            {% endif %}
+            <li class="breadcrumb-item"><a href="{{get_homepath() }}/course/{{ course.get_id() }}">{{ course.get_name(user_manager.session_language()) }} <span class="sr-only">{{ _("(current)") }}</span></a></li>
+            <li class="breadcrumb-item active"><a href="#"><i class="fa fa-cogs"></i> {{ _("Custom fields") }}</a></li>
+        </ol>
+    </nav>
+{% endblock %}
+{% block content %}
+    <h2>{{_("Custom values")}}</h2>
+    <form method="post" action="">
+    {% for key in custom_fields %}
+        <div class="form-group row">
+            <label class="col-sm-2 " for="{{ key }}" class="control-label">{{ key }}</label>
+            {% if  fieldtypes[custom_fields[key].get_type()] == fieldtypes['BOOLEAN'] %}
+                <label><input class="" type="checkbox" id="{{ key }}" name="{{ key }}" {% if custom_values_user[key]=="on" %} checked {% endif %}>{{ custom_fields[key].get_description() }}</label>
```

Comment:
So a boolean may be expected to be ``"on"``. This single line betrays you: there is actually no data check neither in your code. It should be used for more than rendering, because you won't be able to render correctly if data are not the expected one.